pkg/infinity/jsonBackend.go

```diff
@@ -38,11 +39,15 @@ func GetJSONBackendResponse(ctx context.Context, urlResponseObject any, query mo
 		RootSelector: query.RootSelector,
 		Columns:      columns,
 	})
+
+	if err != nil {
+		if errors.Is(err, jsonframer.ErrInvalidRootSelector) || errors.Is(err, jsonframer.ErrInvalidJSONContent) || errors.Is(err, jsonframer.ErrInvalidJSONContent) {
+			return frame, errorsource.DownstreamError(fmt.Errorf("error converting json data to frame: %w", err), false)
+		}
+		return frame, errorsource.PluginError(fmt.Errorf("error converting json data to frame: %w", err), false)
+	}
```

Comment:
I have tried to test this, but I don't think this works. I am still getting plugin error when I write incorrect Rows/Root query: 

<img width="1915" alt="image" src="https://github.com/user-attachments/assets/461e79b6-ade8-4e1c-a4f1-65d03519f21a">
<img width="1509" alt="image" src="https://github.com/user-attachments/assets/337daf83-5c15-40a1-90d9-74e5774a8883">

Reply:
I've had a closer look and should we also consider these as downstream: https://github.com/grafana/infinity-libs/blob/c9bd79b3ed2330cc16dcdf818b05dbc42b8355c2/lib/go/jsonframer/errors.go#L6?

Reply:
We can do that in next iteration. This is currently better than it was before.